vlm/src/test/scala/geotrellis/contrib/vlm/spark/RasterSummarySpec.scala

```diff
@@ -124,4 +126,57 @@ class RasterSummarySpec extends FunSpec with TestEnvironment with BetterRasterMa
 
     contextRDD.stitch.tile.band(0).renderPng().write("/tmp/raster-source-contextrdd.png")
   }
+
+  it("should collect temporal contextRDD") {
+    val dates = "2018-01-01" :: "2018-02-01" :: "2018-03-01" :: Nil
+    val expectedDates = dates.map { str =>
+      val Array(y, m, d) = str.split("-").map(_.toInt)
+      ZonedDateTime.of(y, m, d,0,0,0,0, ZoneOffset.UTC).toInstant.toEpochMilli
+    }
+    val files = dates.map { str => GeoTiffDataPath(Resource.path(s"img/aspect-tiled-$str.tif")) }
+    val targetCRS = WebMercator
+    val method = Bilinear
+    val layoutScheme = ZoomedLayoutScheme(targetCRS, tileSize = 256)
+
+    // read sources
+    val sourceRDD: RDD[RasterSource] =
+      sc.parallelize(files, files.length)
+        .map(uri => GeoTiffRasterSource(uri).reproject(targetCRS, method = method): RasterSource)
+        .cache()
+
+    // function to lift SpatialKeys to SpaceTimeKeys during the metadata collection and during tiling
+    def keyTransform: (RasterSource, SpatialKey) => SpaceTimeKey = (rs, key) => SpaceTimeKey(key, {
+      val date = raw"(\d{4})-(\d{2})-(\d{2})".r.findFirstMatchIn(rs.dataPath.toString)
+      val Some((y, m, d)) = date.map { d => (d.group(1).toInt, d.group(2).toInt, d.group(3).toInt) }
+
+      TemporalKey(ZonedDateTime.of(y, m, d, 0, 0, 0, 0, ZoneOffset.UTC).toInstant.toEpochMilli)
+    })
+
+    // collect raster summary
+    val summary = RasterSummary.fromRDD(sourceRDD, keyTransform)
```

Comment:
To collect `RasterSummary[SpatialKey]` we can use `RasterSummary.fromRDD(sourceRDD)`, for temporal and other `K` types we can just add a `keyTransform` function definition.